### PR TITLE
Make MessageProducerClient idempotent

### PR DESCRIPTION
**Pull request by allanmckenzie**

### CODE REVIEW

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java [hunk 1]
```diff
@@ -40,32 +41,48 @@ public MessageProducerClient() {
...
+
         try {
-            activeMQConnectionFactory.setBrokerURL(QUEUE_URI);
-            connection = activeMQConnectionFactory.createConnection();
-            connection.start();
+            if (connection == null) {
+                createConnection();
+            }
+
+            if(messageProducer != null) {
```
Comment (Morbu) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java:60]: formatting

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java [hunk 2]
```diff
@@ -86,8 +103,7 @@ public void sendMessage(final String commandName, final JsonEnvelope jsonEnvelop
             throw new RuntimeException("Message producer not started. Please call startProducer(...) first.");
         }
 
-        @SuppressWarnings("deprecation")
-        final String json = jsonEnvelope.toDebugStringPrettyPrint();
+        @SuppressWarnings("deprecation") final String json = jsonEnvelope.toDebugStringPrettyPrint();
```
Comment (Morbu) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java:106]: Should this be in the catch? We only use the debug string when we have an exception?

Comment (allanmckenzie) [reply test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java:106]: No - the debug string is used to create the TextMessage Object that is sent

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java (hunk 1 above)
Comment (allanmckenzie) [reply test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java:60]: It is formatted - I reformatted before pushing. It just looks a bit weird here

Comment (Morbu) [reply test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java:60]: if(messageProdu... looks wrong, it looks ok in the IDE ?

**Review by Morbu — APPROVED** [review]

**Pull request merged**